### PR TITLE
Make zookeeper connection string optional

### PR DESCRIPTION
**Pull request by mohitsoni**
This PR makes configuration of zookeeper connection string optional for both YAML and Java interfaces. By default, the value of zookeeper connection string is "master.mesos:2181". This value can be over-ridden in the YAML interface via:

```yaml
...
zookeeper: custom.master.mesos:2181
...
```

or in Java interface via:

```java
new DefaultService(DefaultServiceSpec.newBuilder()
                    .zookeeperConnection("custom.master.mesos:2181")
```

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java [hunk 1]
```diff
@@ -116,7 +119,8 @@ public int getApiPort() {
 
     @Override
     public String getZookeeperConnection() {
-        return zookeeperConnection;
+        // If no zookeeperConnection string is configured, fallback to the default value.
+        return StringUtils.isNotBlank(zookeeperConnection) ? zookeeperConnection : DEFAULT_ZK_CONNECTION;
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java]: Maybe just use isBlank() and switch the cases? Could see it being slightly easier to parse

File: sdk/scheduler/src/test/resources/valid-minimal.yml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 name: "hello-world"
 principal: "hello-world-principal"
-zookeeper: master.mesos:2181
```
Comment (nickbp) [inline sdk/scheduler/src/test/resources/valid-minimal.yml]: Maybe remove default cases from the other .yml's as well

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java (hunk 1 above)
Comment (mohitsoni) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java]: Sure

File: sdk/scheduler/src/test/resources/valid-minimal.yml (hunk 2 above)
Comment (mohitsoni) [inline sdk/scheduler/src/test/resources/valid-minimal.yml]: Sure

Comment (mohitsoni) [inline sdk/scheduler/src/test/resources/valid-minimal.yml]: Done

**Pull request merged**